### PR TITLE
Added quantile regression loss function for some of the more popular quantiles

### PR DESCRIPTION
**Pull request by sachinruk**
The tilted (pinball) loss function is used in quantile regression. Thought it would be useful for keras to implement atleast a few of those quantiles as shown below in objectives.py. I've wrapped the quantiles 1%, 5%, 10%, 50%, 90%, 95% and 99% as q1, q5 etc.

The tilted loss function shown here: https://www.lokad.com/pinball-loss-function-definition can be summarised as mean(q*err + clip(-err)) where err = observation - quantile_function.

### CODE REVIEW

File: keras/objectives.py [hunk 1]
```diff
@@ -63,6 +63,36 @@ def cosine_proximity(y_true, y_pred):
     y_pred = K.l2_normalize(y_pred, axis=-1)
     return -K.mean(y_true * y_pred, axis=-1)
 
+# The actual pinball/ tilted loss function for quantile regression
+def tilted_loss(q, y, f):
```
Comment (fchollet) [inline keras/objectives.py]: Syntax should be `tilted_loss(y_true, y_pred, quantile)`. Unclear that `titled_loss` is the optimal name here; how is it called in sklearn?

File: keras/objectives.py [hunk 2]
```diff
@@ -63,6 +63,36 @@ def cosine_proximity(y_true, y_pred):
...
+    """
+    Pinball loss function used in quantile regression where q is the quantile.
```
Comment (fchollet) [inline keras/objectives.py]: The docstring should be more descriptive of what the loss does.

File: keras/objectives.py [hunk 3]
```diff
@@ -63,6 +63,36 @@ def cosine_proximity(y_true, y_pred):
...
+    """
+    e = (y-f)
+    return K.mean(q*e + K.clip(-e, K.epsilon(), np.inf), axis=-1)
```
Comment (fchollet) [inline keras/objectives.py]: Respect PEP8 conventions.

File: keras/objectives.py [hunk 4]
```diff
@@ -63,6 +63,36 @@ def cosine_proximity(y_true, y_pred):
...
+    return tilted_loss(0.50, y_true, y_pred)
+
+def q90(y_true, y_pred):
+    return tilted_loss(0.90, y_true, y_pred)
+
+def q95(y_true, y_pred):
+    return tilted_loss(0.95, y_true, y_pred)
+
+def q99(y_true, y_pred):
+    return tilted_loss(0.99, y_true, y_pred)
```
Comment (fchollet) [inline keras/objectives.py]: None of these should be included.

File: keras/objectives.py (hunk 1 above)
Comment (sachinruk) [inline keras/objectives.py]: Will get onto fixing syntax. But not sure if this loss function is implemented in sklearn, nor do I know where to look (sorry). This is just equation 1 of http://www-personal.acfr.usyd.edu.au/f.ramos/Fabio_Ramos_Homepage/Publications_files/GPquantile.pdf (it is my paper but its a fairly standard loss for quantile regression).

To some extent it will give you an uncertainty estimate of the predicted function (by taking say the 5th and 95th percentile).

File: keras/objectives.py (hunk 3 above)
Comment (sachinruk) [inline keras/objectives.py]: I fixed as much as I could with the automatic fixer as pointed out on Contributions.md but I'm not too familiar with PEP8. Is there a particular convention that I have violated?

File: keras/objectives.py (hunk 4 above)
Comment (sachinruk) [inline keras/objectives.py]: ok, but can loss functions have more than two inputs? That's why I even had these in the first place. i.e. I was hoping users would implement it as `model.compile(loss="q5", optimizer='adagrad')`.

**Comment by fchollet** [comment]
Closing outdated PR. If you still care about the content of the PR, please submit a new PR to `master`, updated for the Keras 2.0 API.
